pkg/plugin/processor/builtin/impl/webhook/http_test.go

```diff
@@ -342,6 +356,99 @@ func TestHTTPProcessor_Success(t *testing.T) {
 	}
 }
 
+func TestHTTPProcessor_URLTemplate(t *testing.T) {
+	respBody := []byte("foo-bar/response")
+
+	tests := []struct {
+		name     string
+		config   map[string]string
+		pathTmpl string // will be attached to the URL
+		path     string // expected result of the pathTmpl
+		args     []opencdc.Record
+		want     []sdk.ProcessedRecord
+	}{
+		{
+			name:     "URL template, success",
+			config:   map[string]string{"request.method": "GET"},
+			pathTmpl: "/{{.Payload.After.foo}}",
+			path:     "/123",
+			args: []opencdc.Record{{
+				Payload: opencdc.Change{
+					Before: nil,
+					After: opencdc.StructuredData{
+						"foo": 123,
+					},
+				},
+			}},
+			want: []sdk.ProcessedRecord{sdk.SingleRecord{
+				Payload: opencdc.Change{
+					After: opencdc.RawData(respBody),
+				},
+			},
+			},
+		},
+		{
+			name:     "URL template, key with a hyphen",
+			config:   map[string]string{},
+			pathTmpl: `/{{index .Payload.After "foo-bar"}}`,
+			path:     "/baz",
+			args: []opencdc.Record{{
+				Payload: opencdc.Change{
+					After: opencdc.StructuredData{
+						"foo-bar": "baz",
+					},
+				},
+			}},
+			want: []sdk.ProcessedRecord{sdk.SingleRecord{
+				Payload: opencdc.Change{
+					After: opencdc.RawData(respBody),
+				},
+			}},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			is := is.New(t)
+
+			wantMethod := tc.config["request.method"]
+			if wantMethod == "" {
+				wantMethod = "POST" // default
+			}
```

Comment:
Nitpick: if we're focusing on the URL template (as the name of the test suggests), we shouldn't be concerned with the request method, request body or response body. That's for other tests to assert, so this one can be much more focused.